tests/CMakeLists.txt

```
@@ -57,20 +56,22 @@ add_definitions(-DCATCH_CONFIG_FAST_COMPILE=1)
 
 set(common_sources catch/main.cpp ${library_dir}/canard.c ${library_dir}/canard_dsdl.c)
 
-function(gen_test name files compile_definitions compile_flags link_flags)
+function(gen_test name files compile_definitions compile_flags link_flags c_standard)
     add_executable(${name} ${common_sources} ${files})
     target_compile_definitions(${name} PUBLIC ${compile_definitions})
     target_link_libraries(${name} pthread)
-    set_target_properties(${name} PROPERTIES COMPILE_FLAGS "${compile_flags}" LINK_FLAGS "${link_flags}")
+    set_target_properties(${name} PROPERTIES COMPILE_FLAGS "${compile_flags}" LINK_FLAGS "${link_flags}" C_STANDARD "${c_standard}")
     add_test("run_${name}" "${name}" --rng-seed time)
 endfunction()
 
 function(gen_test_matrix name files compile_definitions)
-    gen_test("${name}_x64" "${files}" "${compile_definitions}" "-m64" "-m64")
-    gen_test("${name}_x32" "${files}" "${compile_definitions}" "-m32" "-m32")
+    gen_test("${name}_x64_c99" "${files}" "${compile_definitions}" "-m64" "-m64" "99")
+    gen_test("${name}_x32_c99" "${files}" "${compile_definitions}" "-m32" "-m32" "99")
+    gen_test("${name}_x64_c11" "${files}" "${compile_definitions}" "-m64" "-m64" "11")
+    gen_test("${name}_x32_c11" "${files}" "${compile_definitions}" "-m32" "-m32" "11")
     # Coverage is only available for GCC builds.
     if ((CMAKE_CXX_COMPILER_ID STREQUAL "GNU") AND (CMAKE_BUILD_TYPE STREQUAL "Debug"))
-        gen_test("${name}_cov" "${files}" "${compile_definitions}" "-g -O0 --coverage" "--coverage")
+        gen_test("${name}_cov" "${files}" "${compile_definitions}" "-g -O0 --coverage" "--coverage" "99")
```

Let's stick to C11 as default here.

```suggestion
        gen_test("${name}_cov" "${files}" "${compile_definitions}" "-g -O0 --coverage" "--coverage" "11")
```